tests/checkpointing/test_model_checkpoint.py

```diff
@@ -237,35 +240,55 @@ def configure_optimizers(self):
     ckpt_files = list(Path(tmpdir).glob('*.ckpt'))
     scores = [metric[monitor] for metric in trainer.dev_debugger.logged_metrics if monitor in metric]
     lr_scheduler_debug = trainer.dev_debugger.saved_lr_scheduler_updates
-    assert len(ckpt_files) == len(scores) == per_epoch_call_count * max_epochs
+
+    # on_train_end ckpt callback is called which creates an additional ckpt in case no ckpt is created at the
+    # end of epoch, thus if val_check_interval doesn't align with the training steps we create an additional ckpt
+    additional_ckpt, additional_ckpt_path = 0, None
+    if not epoch_aligned:
+        additional_ckpt_path = [f for f in ckpt_files if 'v1' in f.stem][0]
+        additional_ckpt = 1
+
+    additional_ckpt = 1 if not epoch_aligned else 0
+    assert len(ckpt_files) == len(scores) + additional_ckpt == per_epoch_call_count * max_epochs + additional_ckpt
     assert len(lr_scheduler_debug) == max_epochs
 
+    def _make_assertions(epoch, ix, add=''):
+        global_ix = ix + per_epoch_call_count * epoch
+        score = scores[global_ix]
+        expected_score = getattr(model, f'{monitor}s')[global_ix].mean().item()
+        expected_filename = f'{monitor}={score:.4f}-epoch={epoch}{add}.ckpt'
+        assert math.isclose(score, expected_score, rel_tol=1e-4)
+
+        chk = pl_load(os.path.join(checkpoint.dirpath, expected_filename))
+        assert chk['epoch'] == epoch + 1
+        assert chk['global_step'] == per_epoch_steps * (global_ix + 1) + (left_over_steps * (epoch + (1 if add else 0)))
```

Comment:
nit: just for sanity, can you use some variables for the right side of this assertion?

Reply:
done.

Reply:
@rohitgr7 pls when done, mar it also as resolved :]